extensions/amp-content-viewer/0.1/amp-content-viewer.js

```diff
@@ -0,0 +1,696 @@
+
+    const sourceAspectRatio = this.sourceWidth_ / this.sourceHeight_;
+    let height = Math.min(this.elementBox_.width / sourceAspectRatio,
+        this.elementBox_.height);
+    let width = Math.min(this.elementBox_.height * sourceAspectRatio,
+        this.elementBox_.width);
+
+    if (Math.abs(width - this.sourceWidth_) <= 16
+    && Math.abs(height - this.sourceHeight_ <= 16)) {
+      width = this.sourceWidth_;
+      height = this.sourceHeight_;
+    }
+
+    this.contentBox_ = layoutRectLtwh(
+        Math.round((this.elementBox_.width - width) / 2),
+        Math.round((this.elementBox_.height - height) / 2),
+        Math.round(width),
+        Math.round(height));
+
+    // Adjust max scale to at least fit the screen.
+    const elementBoxRatio = this.elementBox_.width /
+     this.elementBox_.height;
+    const maxScale = Math.max(
+        elementBoxRatio / sourceAspectRatio,
+        sourceAspectRatio / elementBoxRatio
+    );
+    this.maxScale_ = Math.max(DEFAULT_MAX_SCALE, maxScale);
+
+    // Reset zoom and pan.
+    this.startScale_ = this.scale_ = 1;
+    this.startX_ = this.posX_ = 0;
+    this.startY_ = this.posY_ = 0;
+    this.updatePanZoomBounds_(this.scale_);
+  }
+
+  /**
+   * Measures and resets the content dimensions, after the element
+   * dimensions changes.
+   * @return {!Promise}
+   */
+  resetContentDimensions_() {
+    return this.measureElement(() => this.measure_()).then(() => {
+      const content = dev().assertElement(this.content_);
+      return this.mutateElement(() => {
+        // Set the actual dimensions of the content
+        st.setStyles(content, {
+          top: st.px(this.contentBox_.top),
+          left: st.px(this.contentBox_.left),
+          width: st.px(this.contentBox_.width),
+          height: st.px(this.contentBox_.height),
+        });
+        // Update translation and scaling
+        this.updatePanZoom_();
+      }, content);
+    });
+  }
+
+  /** @private */
+  cleanupGestures_() {
+    if (this.gestures_) {
+      this.gestures_.cleanup();
+      this.gestures_ = null;
+    }
+  }
+
+  /** @private */
+  setupGestures_() {
+    // TODO (#12881): this and the subsequent use of event.preventDefault
+    // is a temporary solution to #12362. We should revisit this problem after
+    // resolving #12881 or change the use of window.event to the specific event
+    // triggering the gesture.
+    this.gestures_ = Gestures.get(
+        this.element,
+        /* opt_shouldNotPreventDefault */ false
+    );
+
+    this.gestures_.onPointerDown(() => {
+      if (this.motion_) {
+        this.motion_.halt();
+      }
+    });
+
+    // Zoomable.
+    this.gestures_.onGesture(DoubletapRecognizer, e => {
+      const newScale = this.scale_ == 1 ? this.maxScale_ : this.minScale_;
+      const deltaX = this.elementBox_.width / 2 - e.data.clientX;
+      const deltaY = this.elementBox_.height / 2 - e.data.clientY;
+      this.onZoom_(newScale, deltaX, deltaY, true).then(() => {
```

Comment:
nit: for boolean arguments, it's helpful to annotate what the name of the argument is `/* animate */ true);`

Reply:
done